src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -326,6 +321,59 @@ static bool matchMetadata(BSONObj& md1, BSONObj& md2)
 static bool equalMetadataValues(BSONObj& md1, BSONObj& md2)
 {
 
+
+  // Same declared type ?
+  if (getField(md1, ENT_ATTRS_MD_TYPE).type()!= getField(md2, ENT_ATTRS_MD_TYPE).type())
+  {
+    return false;
+  }
+  switch (getField(md1, ENT_ATTRS_MD_TYPE).type())
+  {
+    /* FIXME not yet
+    case Object:
+      ...
+      break;
+
+    case Array:
+      ...
+      break;
+    */
+
+    case NumberDouble:
+      if (getField(md1, ENT_ATTRS_MD_TYPE).Number() != getField(md2, ENT_ATTRS_MD_TYPE).Number())
+      {
+        return false;
+      }
+      break;
+
+    case Bool:
+      if (getBoolField(md1, ENT_ATTRS_MD_TYPE) != getBoolField(md2, ENT_ATTRS_MD_TYPE))
+      {
+        return false;
+      }
+      break;
+
+    case String:
+      if (getStringField(md1, ENT_ATTRS_MD_TYPE) != getStringField(md2, ENT_ATTRS_MD_TYPE))
+      {
+        return false;
+      }
+      break;
+
+    case jstNULL:
+      if (!md2.getField(ENT_ATTRS_MD_TYPE).isNull())
+      {
+        return false;
+      }
+      break;
+
+    default:
+      LM_E(("Runtime Error (unknown metadata type type in DB: %d)", getField(md1, ENT_ATTRS_MD_TYPE).type()));
```

Comment:
Yeah, sounds better. Or maybe:

"unknown JSON type for metadata NGSI type"

achieves more precision.